web/src/components/customer/contexts/CustomerContext.tsx

```diff
@@ -21,11 +21,16 @@ import useLocalStorage from 'components/common/hooks/useLocalStorage';
 import {Customer} from 'interfaces';
 import {getIdentity} from 'microapps';
 
+interface AuthenticatedCustomer {
+  customer: Customer | undefined;
+  idToken: string | undefined;
+};
+
 type ContextType =
   | {
       idToken: string | undefined;
       customer: Customer | undefined;
-      getCustomerWithLogin: () => Promise<Customer | undefined>;
+      getCustomerWithLogin: (x?: boolean) => Promise<AuthenticatedCustomer>;
```

Comment:
what is this "x?: boolean"? can you rename x?

Reply:
@ifhuang Oops this `x` is something that I was using for debugging, forgot to remove it from this line! Thanks for pointing this out!